source/docs/essentials/create-a-business-component.md

```diff
@@ -0,0 +1,461 @@
+          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+        />
+        <Marker position={coordinates}>
+          <Popup>
+            <div>
+              My awesome store ─ {props.store.name}
+              Email: {props.store.owner.email}
+              Phone: {props.store.phone}
+            </div>
+          </Popup>
+        </Marker>
+      </Map>
+    </div>
+  );
+};
+
+StoreLocator.propTypes = {
+  loading: PropTypes.bool.isRequired,
+  store: PropTypes.shape({
+    name: PropTypes.string.isRequired,
+    phone: PropTypes.string.isRequired,
+    owner: PropTypes.shape({
+      email: PropTypes.string.isRequired
+    }).isRequired,
+    coordinates: PropTypes.shape({
+      longitude: PropTypes.number.isRequired,
+      latitude: PropTypes.number.isRequired
+    }).isRequired
+  })
+};
+
+// Let's not forget to use the Enhancer
+export default EnhanceStoreLocator(StoreLocatoreQuery)(StoreLocator);
+```
+
+<blockquote class="note">
+About **PropTypes**: they allow us to validate the props passed to a child
+component, this will log explicit errors if you pass invalid props in dev mode.
+This will also serve as documentation for other developers in your team. They may
+be a little verbose, but maintaining them will help you in the long run.
+</blockquote>
+
+## Using it in our App
+
+We now have extracted all the store locator logic. We can now use our
+brand new and shiny module component within the homepage.
+
+```js
+// my-module/theme/pages/Home/Home.js;
+// ...
+import StoreLocator from "theme/modules/StoreLocator";
+
+const Home = ({ store }) => (
+  <div className="page page--home">
+    {/* ... */}
+    <StoreLocator />
+  </div>
+);
+```
+
+As you can see, we did not use a relative import. This is because in
+Front-Commerce we have a few aliases that will let you import files without
+worrying about your current position in the folder structure.
+
+In our case, the `Home` component being in
+`my-module/web/theme/pages/Home/Home.js`, we do not have to import the
+`StoreLocator` by using relative paths `../../modules/StoreLocator` but we
+can remain with `theme/modules/StoreLocator` which is more explicit. This
+is possible for any file located in the folder `web/theme` of a module.
+
+And it works! You now have a clean `Home` page component that uses a Business
+component which could be used anywhere in your application (About us,
+Contact, etc.).
+
+<blockquote class="note">
+**Going further:** The store locator we just created is very simple and it
+has a very limited Business impact. The store locator module does not need
+to know the implementation of the map itself (like the fact of using
+`react-leaflet`). So a map component could be extracted in a **UI** component.
+But for the sake of this guide, we kept it simple.
+</blockquote>
+
+As a final note, please keep in mind that splitting code is a difficult task.
```

Comment:
```suggestion
## A word about breaking up components

As a final note, please keep in mind that splitting code is a difficult task.
```

I think it deserves an explicit title to highlight this advice.

Do you think it would be interesting to also add a link to https://blog.kentcdodds.com/when-to-break-up-a-component-into-multiple-components-4ee75ab53bbc ?